test/mini_racer_test.rb

```diff
@@ -1067,9 +1065,26 @@ def test_regexp_string_iterator
       # TODO(bnoordhuis) maybe detect the iterator object and serialize
       # it as an array of strings; problem is there is no V8 API to detect
       # regexp string iterator objects
-      assert_match(/\[object RegExp String Iterator\] could not be cloned/, e.message)
+      assert_match(
+        /\[object RegExp String Iterator\] could not be cloned/,
+        e.message
+      )
       exc = true
     end
     assert exc
   end
+
+  def test_function_cloning
+    message = nil
+    context = MiniRacer::Context.new
+    context.attach("log", proc { |msg| message = msg })
+    context.eval <<~JS
+       console = {
+         log: function(...args){ log(args.join(" ")); },
+       }
```

Comment:
I see what the problem is: the result of this statement is the `log` function object. The test passes when changed to this:
```suggestion
       console = {
         log: function(...args){ log(args.join(" ")); },
       }
       undefined
```
Let me think what a good solution is.

Reply:
For completeness sake, another workaround is to make the property non-enumerable:
```suggestion
       console = {}
       Object.defineProperty(console, "log", {value: function(...args){ log(args.join(" ")); }})
```

Reply:
For posterity, I think there are 4 options:

1. do nothing and let the error bubble up (current situation)
2. detect the error and return something else, like an empty object or a marker
3. deep-copy the object and replace or drop function properties
4. ditto but only shallow-copy the object, i.e., fix up only one level

3 is O(n) time and space, 4 is only a partial fix, so I guess 2 is the best option.